src/scripts/analyse-nala.ts

```diff
@@ -0,0 +1,243 @@
+import { walk } from './common'
+import path from 'path'
+import fs from 'fs'
+import { execSync } from 'child_process'
+
+interface Options {
+  extractTimestamp: boolean
+}
+
+interface UsageMatch {
+  type: 'color' | 'spacing' | 'font'
+  raw: string
+  property: string
+  value: string
+  isNala: boolean
+  file: string
+  line: number
+  column: number
+  timestamp?: number
+}
+
+const allowedExtensions = [
+  '.html',
+  '.svelte',
+  '.css',
+  '.scss',
+  '.sass',
+  '.less',
+  '.lss',
+  '.js',
+  '.ts',
+  '.jsx',
+  '.tsx'
+]
+
+const spacingPatterns = [
+  /\spadding(-(top|right|bottom|left))?:/,
+  /\smargin(-(top|right|bottom|left))?:/,
+  /\s(gap|row-gap|column-gap):/,
+  /\s(top|right|bottom|left):/,
+  /\sinset(-(top|right|bottom|left))?:/
+]
+
+// There's a bunch of stuff we'll miss here (like setting just 'background')
+const colorPatterns = [
+  /\scolor:/,
+  /\sbackground-color:/,
+  /\sborder-color:/,
+  /\soutline-color:/,
+  /\sfill:/,
+  /\sstroke:/
+]
+
+const fontPatterns = [/\sfont((\w|-)+)?:/, /\sline((\w|-)+)?:/]
+
+const detectors = {
+  color: colorPatterns,
+  spacing: spacingPatterns,
+  font: fontPatterns
+}
+
+const isNalaValue = (value: string) => {
+  return (
+    value.includes('--leo-') ||
+    value.includes('color.') ||
+    value.includes('font.') ||
+    value.includes('spacing.')
+  )
+}
+
+const isFileTracked = (filepath: string) => {
+  // TODO: we shouldn't use try/catch here as it'll be slow. Instead we should
+  // check the exit code but I'm in a rush.
+  try {
+    execSync(`git ls-files --error-unmatch -- ${filepath}`, { stdio: 'ignore' })
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+const getLineDateLookup = (filepath: string): number[] | undefined => {
+  try {
+    const content = execSync(`git blame --line-porcelain -- ${filepath}`, {
+      encoding: 'utf8'
+    })
```

Comment:
```suggestion
```suggestion
    const content = execSync('git', ['blame', '--line-porcelain', '--', filepath], {
      encoding: 'utf8'
    })
```
```